### DISCUSSION TITLE
Question on RemoteGraphicsView

### DISCUSSION THREAD

**Discussion by matthijs-chainic** [pyqtgraph/pyqtgraph#2118] [Q&A] [answered]
Hi,

For a while now I've been struggling to get a `RemoteGrahicsView` window to work. I want to create a window from my application that runs in a different process, to not slow down the main program. I recreated my problem:

```python
import sys

from PySide6.QtWidgets import QApplication, QMainWindow, QCheckBox
import pyqtgraph as pg


class Application(QApplication):
    def __init__(self):
        super(Application, self).__init__(sys.argv)

        self.window = QMainWindow()

        # Test for responsiveness
        self.window.placeholder = QCheckBox(self.window)
        self.plotWindow = Foo()
        self.window.show()


class Foo(pg.RemoteGraphicsView):
    def __init__(self):
        super(Foo, self).__init__(debug=False)

        self.layout = pg.GraphicsLayout()

        self.button = QCheckBox(self)  # Test for responsiveness

        self.p = self.layout.addPlot(0, 0)
        self.p.plot([1, 2, 3, 4, 5])

        self.show()


if __name__ == "__main__":
    app = Application()
    sys.exit(app.exec())
```

Both windows are responsive (the checkboxes work), but the plot does not appear. I hope someone can point out my error.

![image](https://user-images.githubusercontent.com/92378596/143062643-51b5a397-3f95-4373-90f7-50835ee8efc7.png)

The [documentation on `RemoteGraphicsView`](https://pyqtgraph.readthedocs.io/en/latest/widgets/remotegraphicsview.html) is not helping me a lot. I know of proxies, but I have no idea what is meant with the following line:

> GraphicsItems must be created by proxy to the remote process.

I am not sure if this is related to my problem, though.

Thanks for any help!

**Comment by pijyoi** [comment, marked as answer]
Taking reference from ```pyqtgraph/examples/RemoteGraphicsView.py```, your example would be modified to:
```python
import pyqtgraph as pg
from pyqtgraph.Qt import QtWidgets

class MainWindow(QtWidgets.QMainWindow):
    def __init__(self):
        super().__init__()
        self.setCentralWidget(QtWidgets.QCheckBox())
        self.show()
        self.foo = Foo()

class Foo(QtWidgets.QWidget):
    def __init__(self):
        super().__init__()

        self.rgfxview = pg.RemoteGraphicsView(debug=False)

        layout = QtWidgets.QVBoxLayout()
        self.button = QtWidgets.QCheckBox()
        layout.addWidget(self.button)
        layout.addWidget(self.rgfxview)
        self.setLayout(layout)

        self.rplt = self.rgfxview.pg.PlotItem() 
        self.rgfxview.setCentralItem(self.rplt)
        self.rplt.plot([1, 2, 3, 4, 5])

        self.show()

pg.mkQApp()
main = MainWindow()
pg.exec()
```

>GraphicsItems must be created by proxy to the remote process.

This probably means what ```self.rplt = self.rgfxview.pg.PlotItem()``` is doing. In order to instantiate a remote ```PlotItem```, rather than doing the usual ```pg.PlotItem()```, you do ```rgfxview.pg.PlotItem()```.

The question I have for you is, do you already have an application where not using ```RemoteGraphicsView``` is too slow for your application? Have you tried using the ```master``` branch?

**Reply by matthijs-chainic** [reply]
Thank you so much for the support!

> The question I have for you is, do you already have an application where not using `RemoteGraphicsView` is too slow for your application?

Yeah, I think I do. I'm not 100% certain this is slowing it down, but I have plots of around 1-2 million datapoints, and the application keeps growing in complexity. Even with a lower amount of datapoints, I have noticed that the responsiveness of the `MainWindow` has decreased (delay in input).

> Have you tried using the master branch?

No, I am using the [latest release](https://github.com/pyqtgraph/pyqtgraph/releases/tag/pyqtgraph-0.12.3).

Thanks again!
